### PR TITLE
Processing data

### PR DESCRIPTION
**Pull request by jaimevillegas**
Hi there!

I have made the following changes to the repository:

- Add `user.rb` model.
- add `like.rb` model.
- add `post.rb` model.
- add `comment.rb` model.
- Create custom methods according to project requirements.
- Test everything using the Rails interactive console.

Thank you for taking the time to review my code! ⚡ 🚀

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+  has_many :likes
+
```
Comment (rloterh) [inline app/models/user.rb]: - [x] With the given [_ERD_](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png), whiles establishing a _has many_ **association** for the _User_ model, we need to consider that _Author_ is aliasing _User_ model and therefore  _author_id_ should be referenced as **foreign key** in _Post_, _Comment,_ and _Like_ models. Kindly ensure to improve upon this aspect, and our code should be similar to:
```ruby
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :user, optional: true
```
Comment (rloterh) [inline app/models/post.rb]: - [x] As per the given _ERD_, establishing the _Post_ **association** should be such that _Post_ belong to _User_ **through** _Author_, an alias of _User_, like below:
`belongs_to :author, class_name: 'User'` 

Let's also note that this association is **not optional**

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require_relative 'application_record'
+
+class Comment < ApplicationRecord
+  belongs_to :user, optional: true
+  belongs_to :post, optional: true
```
Comment (rloterh) [inline app/models/comment.rb]: - [x] Simlarly, the _Comment_ **association** should be such that _Comment_ belong to _User_ **through** _Author_, an alias of _User_, like below:
`belongs_to :author, class_name: 'User'` 

  Let's note that non of the associations established with the _comment_ model are optional.

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class Like < ApplicationRecord
+  belongs_to :user, optional: true
+  belongs_to :post, optional: true
```
Comment (rloterh) [inline app/models/like.rb]: - [ ] 
- [x] Again, as per the given _ERD_, establishing the _Like_ **association** should be such that _Like_ belong to _User_ **through** _Author_, an alias of _User_, like below:
`belongs_to :author, class_name: 'User'

  Here also non of the associations established with the like model are optional.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Jaime, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁 🚦  🚗 _ 🏃 🌴  🏠 🌳 🏘️ 🏡  

## Highlights 👍  
- Good job implementing the `after_save` active callback on the update methods!
- ReadMe looks great!

## Required Changes ♻️
_Kindly also check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by jaimevillegas** [comment]
Hi @rloterh!

Thank you so much for taking the time to review the code and explain to me what I need to improve to meet all the project requirements. I have worked on each comment that you wrote, and now the project is complete! Thank you!

**Review by prolajumokeoni — APPROVED** [review]
Hi @jaimevillegas 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- All changes requested have been implemented 🥇  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**